source/common/router/config_impl.cc

```diff
@@ -324,6 +324,22 @@ void DecoratorImpl::apply(Tracing::Span& span) const {
 
 const std::string& DecoratorImpl::getOperation() const { return operation_; }
 
+RouteTracingImpl::RouteTracingImpl(const envoy::api::v2::route::Tracing& tracing)
```

Comment:
I don't think you are defaulting these to 100%? Can you double check this and add tests if that is what we want to do for defaults? (Or change the docs).

Reply:
Based on your previous comment, I was wondering whether a default of 0% would be appropriate here, as they are specific to the route - however the `overall_sampling` default of 0% does not seem right.

So think it may be better to use 100% for consistency now, and then if the decision is made to change to 0% at a later date, that can be done at both levels.